impl/src/testing.rs

```diff
@@ -21,6 +21,8 @@ use tempfile::TempDir;
 
 use std::{
   collections::HashMap,
+  env,
+  env::{current_dir, set_var},
```

Comment:
can you drop these use statements?  It ends up being only marginally longer, but more readable at the call site.

Reply:
Sure, do you want `std::env::` or `env::` inline? I had `std::env::` at first, but thought it was a bit ugly. No strong preference, though.

Reply:
I'd generally prefer `std::env::`.  Sorry, I'm a bit of a curmudgeon when it comes to this.

Reply:
No worries, will fix. I think your way makes sense verbally, but is a bit visually ugly. 50/50 from my perspective.